src/Control/Carrier/Choose/Church.hs

```diff
@@ -82,5 +84,6 @@ instance MonadTrans ChooseC where
 
 instance Algebra sig m => Algebra (Choose :+: sig) (ChooseC m) where
   alg (L (Choose k)) = ChooseC $ \ fork leaf -> fork (runChoose fork leaf (k True)) (runChoose fork leaf (k False))
-  alg (R other)      = ChooseC $ \ fork leaf -> alg (handle (pure ()) (runChoose (liftA2 (<|>)) (runChoose (liftA2 (<|>)) (pure . pure))) other) >>= runChoose fork leaf
+  alg (R other)      = ChooseC $ \ fork leaf -> alg (handle (pure ()) dst other) >>= runIdentity . runChoose (coerce fork) (coerce leaf) where
+    dst = runIdentity . runChoose (liftA2 (liftA2 (<|>))) (pure . runChoose (liftA2 (<|>)) (pure . pure))
```

Comment:
```suggestion
    dst :: ChooseC Identity (ChooseC m a) -> m (ChooseC Identity a)
    dst = runIdentity . runChoose (liftA2 (liftA2 (<|>))) (pure . runChoose (liftA2 (<|>)) (pure . pure))
```

Reply:
Need some `ScopedTypeVariables` for this, but yeah.

Reply:
Just need an `Applicative m` constraint, as it turns out.